scripts/virtualized_install.sh

```diff
@@ -0,0 +1,481 @@
+        return 1
+    fi
+    
+    log_message "INFO" "Setting up ${service_type^^} proxy for container: $container_name"
+    log_message "INFO" "External port: $proxy_external_port -> Internal: $proxy_url:$proxy_internal_port"
+    
+    sudo lxc config device add "$container_name" "$proxy_name" proxy listen=tcp:0.0.0.0:"$proxy_external_port" connect=tcp:"$proxy_url":"$proxy_internal_port"
+}
+
+# Set up both app and API proxies
+function setup_all_proxies() {
+    set_proxy_for_lxd "app"
+    set_proxy_for_lxd "api"
+}
+
+# overrides the env variables present inside the /etc/nixopus/source/api/.env and /etc/nixopus/source/view/.env files
+# overriding is carried out because the nixopus will be installed in the container, so we need to update the ports and set proxy for the services run inside the container
+function override_env_variables() {
+    local container_name="${CONFIG[container_name]}"
+    local app_domain="${CONFIG[app_domain]}"
+    local api_domain="${CONFIG[api_domain]}"
+
+    local app_env_file="/etc/nixopus/source/view/.env"
+    local api_env_file="/etc/nixopus/source/api/.env"
+
+    log_message "INFO" "Overriding environment variables in container: $container_name"
+    
+    sudo lxc exec "$container_name" -- bash -c "
+        if [[ ! -f \"$app_env_file\" ]]; then
+            echo \"File not found: $app_env_file\"
+            exit 1
+        fi
+        if [[ ! -f \"$api_env_file\" ]]; then
+            echo \"File not found: $api_env_file\"
+            exit 1
+        fi
+
+        sed -i \"s|^WEBSOCKET_URL=.*|WEBSOCKET_URL=wss://$app_domain/ws|\" \"$app_env_file\"
+        sed -i \"s|^WEBHOOK_URL=.*|WEBHOOK_URL=https://$app_domain/api/v1/webhook|\" \"$app_env_file\"
+        sed -i \"s|^API_URL=.*|API_URL=https://$api_domain/api|\" \"$app_env_file\"
+        
+        sed -i \"s|^ALLOWED_ORIGIN=.*|ALLOWED_ORIGIN=https://$app_domain|\" \"$api_env_file\"
+        
+        echo \"Environment variables updated successfully\"
+    "
+}
+
+# Validate installation parameters
+function validate_virtualized_params() {
+    log_message "INFO" "Validating virtualized installation parameters..."
+    local validation_failed=false
+    
+    validate_environment "${CONFIG[env]}" || validation_failed=true
+    validate_lxd || validation_failed=true
+    validate_sudo || validation_failed=true
+    validate_file "${SCRIPT_DIR}/util.sh" "util.sh" || validation_failed=true
+    
+    log_message "INFO" "Checking if ports are available..."
+    is_port_available "${CONFIG[app_port]}" || validation_failed=true
+    is_port_available "${CONFIG[api_port]}" || validation_failed=true
+    
+    if [ -n "${CONFIG[api_domain]}" ]; then
+        validate_domain "${CONFIG[api_domain]}" "API" "false" || validation_failed=true
+    fi
+    if [ -n "${CONFIG[app_domain]}" ]; then
+        validate_domain "${CONFIG[app_domain]}" "App" "false" || validation_failed=true
+    fi
```

Comment:
i think we are using api and app in small cases in other places, wont it be better to continue same convention to avoid confusion?